chirp/drivers/uv5x3.py

```diff
@@ -855,54 +855,72 @@ def apply_freq(setting, obj):
         rs.set_apply_callback(apply_freq, _mem.vfo.b)
         work.append(rs)
 
-        val = bfc.bcd_decode_freq(_mem.subvfoa.vhf.freq)
-        if int(float(val)) == 0:
-            val = str(int(_mem.limits.vhf.lower)) + ".000000"
+        try:
+            val = ""
```

Comment:
Hmm, why do you have these? They're unnecessary (as long as you have the thing on L863 below in the exception handlers. I can clean these up before merge if you want.

Otherwise yeah, this looks much nicer!

Reply:
The reason is because without them, if the there is a bcd coding error, the 'val' from the previous vfo a/b will be used. It bugged that the value from vhf2 would end up in uhf if the image didn't have a uhf value established at the factory. So I guess using something like val1, val2, val3, val4, val5 and val6 could be used to avoid the problem. Do you have a different idea?

Reply:
Erm, I don't see any way that could be true. As soon as L860 runs, this copy of val is gone. If that fails to run, then L863 will wipe it away. In either case, this value is never seen by the first line that actually uses it, which is L864.

I'm wondering if you saw that behavior before you added the `val = "000.000"` in the exception handler of each? Because if you didn't have that, then yeah, a failure in the "VFO B VHF" case would re-use the value from the "VFO A VHF" case. But what you have here can't.

If I'm wrong, can you tell me how to repro? Shoot me an image or something and I'll test loading it with these lines gone and see what you see.

Reply:
You may be right. Let me test it again.